### ISSUE TITLE
I want to run an xfce linux tab on my chromebook rather then switch between xfce and chrome os session. Im new to this and i have a couple of issues i really need help with. 

### ISSUE THREAD

**Issue by TheBadassQueensOliviaAndRegina** [dnschneid/crouton#1960]
Yesterday I used this website: http://www.howtogeek.com/162120/how-to-install-ubuntu-linux-on-your-chromebook-with-crouton/ :to guide me through installing Linux on my Samsung Chromebook. After that ordeal was over i was tired and decided to check it out today. This morning i entered a session and decided to just mess around and get used to using this. First i went on YouTube, and for some reason the videos won't load. Why is that and how do i fix this?  Then when i tried putting the volume up using the keys, the tabs from files bar were dropping down. Is there a way too stop this and give the volume key its normal function when I'm using xfce? After this I logged out of my session to see if there were solutions to this, which is how i stumbled upon this article: http://www.howtogeek.com/208368/how-to-run-a-full-linux-desktop-in-a-browser-tab-on-your-chromebook/ . Since I had already downloaded my Linux i figured i'll just try and add the "xiwi" or "extension". After I put the example they had provide in the crosh terminal thing and entered my password it said that it will begin installing once I log out of all my chroots. So i went back in my session and once again logged out. After this, I tried to download xiwi again and got the same message.  I decided to skip to the next step, and installed this https://chrome.google.com/webstore/detail/crouton-integration/gcpneefbbnfalgjniomfjknbcgkbijom . I also googled "how to download a xiwi extension to an already downloaded xfce crouton?" and that's how I found this website, with someone else having an issue similar to mine. https://github.com/dnschneid/crouton/issues/1301
I've tried all of their codes but i keep getting this message every time---
Password: 
Migrating data from legacy chroots directory /usr/local/chroots to /mnt/stateful_partition/crouton/chroots...
Not unmounting /usr/local/chroots/precise as another instance is using it.
The above chroots appear to be running from the legacy chroots directory.
Log out of all running chroots and the install will automatically continue.
Press Ctrl-C at any time to abort the installation.

---

I also have continuously logged out and nothing. I want to click on the shut down option instead of logout but it does nothing when i click on it. What do I do? Am i going to need to wipe my Chromebook and restart this process again? Is there a way around this? If i do restart should i go back to xfce or try something new?  Btw I downloaded Linux in the first place because i want to edit videos and Chromebook sucks with downloading and the only real option I have to use for video editing is WeVideo. With Linux the are more options. Please, please, please, some one help me, I'm so desperate for something to go right. If they'rs any confusion in this long question please ask so I can clear up any confusion.
Basically these are my questions:
Can I download the xiwi extionsion thing that the website says is needed for the tab to work?
If I can, how?
If I can't do what do I do? 
Should I delete xfce and start the process over?
If that's what it comes down to, should i stick with xfce or are there other "chroots?" that better meet my need for downloading, and video editing?
If I stick with xfce, how do I fix the volume key issues, get YouTube videos to work, and get the "Shut Down" option to work? 
Of course all questions don't need to be answered it just depends on what it comes down to.
Thank you in advanced for all the help, I super, duper appreciate it.

**Comment by DennisLfromGA** [comment]
First of all, I would log out of your Chromebook, shut it down and log back in.
Then open a crosh shell session: 

  <kbd>Ctrl+ALt+t</kbd>
  enter: `shell`

Then proceed with the following from a crosh shell, not in your chroot.

----

> Can I download the xiwi extionsion thing that the website says is needed for the tab to work?
> If I can, how?

To add the 'xiwi' extension to your existing chroot, do this: 

  `sudo sh ~/Downloads/crouton -t xiwi -u`

If you've installed more than one chroot, you'll also need to enter `-n chrootname` to the above command.

> Should I delete xfce and start the process over?

No, that's not necessary.

> If that's what it comes down to, should i stick with xfce or are there other "chroots?" that better meet my need for downloading, and video editing?

You can *add* other targets/DesktopEnvironments to your existing chroot just by updating your chroot and specifying the new target(s) at the same time, you don't need to add a chroot for each DE. For instance, to add 'xiwi' and 'lxde' at the same time do this:

  `sudo sh ~/Downloads/crouton -t xiwi,lxde -u`

Then, to start 'xfce', do this: `sudo startxfce4 [-n chrootname]`
And,   to start 'lxde', do this: `sudo startlxde   [-n chrootname]`

If you want to start 'xfce' in a tab, do this: `sudo startxfce4 -X xiwi-tab [-n chrootname]`

> If I stick with xfce, how do I fix the volume key issues, get YouTube videos to work,...

When using 'xiwi', the Chrome OS volume keys work in the chroot, there's no need to assign new key combos.

> and get the "Shut Down" option to work?

Don't use the 'Shut Down' option, use the 'Log Out' option.

**Comment by TheBadassQueensOliviaAndRegina** [comment]
@DennisLfromGA Thank you so much for your help, but it didn't work. I did everything you said and it seemed like it was downloading the xiwi extension fine but when I thought it was done the message at the end said:
---
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 xserver-xorg-video-dummy : Depends: xorg-video-abi-11
                            Depends: xserver-xorg-core (>= 2:1.10.99.901)
E: Unable to correct problems, you have held broken packages.
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/precise...
---
I thought maybe this is what it's suppose to say and tried opening an xfce tab with the code you wrote, and it didn't work. Is there another code that might work? Im so sorry but I don't know how to figure this out.

**Comment by DennisLfromGA** [comment]
@TheBadassQueensOliviaAndRegina,

Okay, get into your chroot via a command line:

  `sudo enter-chroot`

At this point you may/should be prompted with something like:

> A chroot setup script still exists inside the chroot.
  The chroot may not be fully set up.

> Would you like to finish the setup? [Y/n/d]

Just press ENTER and see if the setup completes without error, hopefully.

If not, you may have to try and fix it another way inside the chroot using:

  `sudo apt-get update; sudo apt-get install -f; sudo apt-get upgrade`

Hopefully, one of the above two methods - or both - should work.

If not, will see you soon... ;)

**Comment by TheBadassQueensOliviaAndRegina** [comment]
Ok I did the first thing and I wasn't sure how to see if it worked so i entered an xfce session. Now the web browser wont open up because of an input/output problem. I'm so sorry for all the trouble. What should i do? is there even a solution to this? lol i must be a pain rite now, im really sorry.

**Comment by DennisLfromGA** [comment]
@TheBadassQueensOliviaAndRegina,

> Ok I did the first thing and I wasn't sure how to see if it worked...

If it completed without displaying any errors, it worked.

> Now the web browser wont open up because of an input/output problem.

Okay, so your xfce desktop works but your web browser won't open?
If that's the case, I would try another crouton update to see if that will fix it using:

  `sudo sh ~/Downloads/crouton -u`

> What Im not to clear about is if i dont have a seperate chroot do i need to add this?

Probably not, to start xfce in a tab just use: `sudo startxfce4 -X xiwi-tab`

-DennisL ( *not* @dnschneid by any stretch of the imagination ;)

**Comment by TheBadassQueensOliviaAndRegina** [comment]
Hello again sorry about the sudden M.I.A., i got busy with family stuff. Anyway, i decided to completely wipe out everything in my chrome book and re-install xfce but with the xiwi included. The 1st 2 times it didn't complete the download, but it worked the third time. Finally i can open xfce as a new tab. Seriously, I'm so happy about this, but once again there's an issue opening the browser. When i click to open it, i get [Failed to execute default Web Browser.] I'm so grateful for how helpful u have been to me, I apologize for the piece of work I've been, like uo have no idea. Thanks tho

**Comment by DennisLfromGA** [comment]
@TheBadassQueensOliviaAndRegina,

Get into a crosh shell and show is the output of: `sudo edit-chroot -al`
That'll let us know what we're dealing with and will help us help you.

**Comment by TheBadassQueensOliviaAndRegina** [comment]
I typed it in and this is what i got:
name: trusty
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/trusty...
crouton: version 1-20150730171703~master:f8fdfb0c
release: trusty
architecture: armhf
xmethod: xiwi
targets: xfce,xiwi
host: version 7077.111.0 (Official Build) stable-channel daisy 
kernel: Linux localhost 3.8.11 #1 SMP Tue Jul 28 14:51:39 PDT 2015 armv7l armv7l armv7l GNU/Linux
freon: yes
Not unmounting /mnt/stateful_partition/crouton/chroots/trusty as another instance is using it.

**Comment by DennisLfromGA** [comment]
> Failed to execute default Web Browser

I found a couple of web sites that discuss it, maybe they'll help:
- [ [Solved] Failed to execute default Web Browser](http://forum.xfce.org/viewtopic.php?pid=26487#p26487)
- [Failed to execute default web browser - input output error](http://ubuntuforums.org/showthread.php?t=2053643)
